paddle/fluid/operators/mkldnn/softmax_mkldnn_op.cc

```diff
@@ -34,47 +34,61 @@ using platform::to_void_cast;
 
 class SoftmaxMKLDNNHandler : public platform::MKLDNNHandler {
  public:
-  SoftmaxMKLDNNHandler(const platform::MKLDNNDeviceContext& dev_ctx,
+  SoftmaxMKLDNNHandler(const std::vector<int>& dims,
+                       framework::proto::VarType::Type vtype,
+                       const mkldnn::memory::data_type dtype,
```

Comment:
As I see you save variable and data type for usage of memory allocation with mutable_data and for memory descriptor creation in MKL-DNN. However mutable_data allows providing a template type as an argument to initialize tensor. For MKL-DNN you can use `platform::MKLDNNGetDataType<T>()` to convert a type to MKL-DNN's data type. Please consider templatizing this handler to handle those scenarios in this way.

Reply:
done